src/transport.ts

```diff
@@ -1,4 +1,5 @@
 import stream  from 'stream';
+import { AsyncResource } from 'async_hooks';
```

Comment:
doesn't this change the minimum node version?
- https://nodejs.org/api/async_context.html#class-asyncresource

<summary style="box-sizing: border-box; margin: 0.5rem 0px; padding: 0.5rem 0px; cursor: pointer;">History</summary><slot id="details-content" style="color: rgb(255, 255, 255); font-family: Lato, &quot;Lucida Grande&quot;, &quot;Lucida Sans Unicode&quot;, &quot;Lucida Sans&quot;, Verdana, Tahoma, sans-serif; font-size: 13.6px; font-style: normal; font-variant-ligatures: none; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(9, 12, 21); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">

Version | Changes
-- | --
v16.4.0 | AsyncResource is now Stable. Previously, it had been Experimental.

</slot>

Reply:
Possibly. Though experimental still means it's available and might work on older versions unless there were breaking changes in it through out the stabilization process.

Reply:
so, this needs to be figured out what is the minimum version. may need engine bumping

Reply:
It looks like it's available all the way back to Node 10: https://nodejs.org/docs/latest-v10.x/api/async_hooks.html#async_hooks_class_asyncresource